### PR TITLE
Add TWT ratio tap changer modification

### PR DESCRIPTION
**Pull request by ayolab**

### CODE REVIEW

File: src/components/dialogs/network-modifications/two-windings-transformer/creation/tap-changer-pane/tap-changer-steps.js [hunk 1]
```diff
@@ -127,27 +167,40 @@ const TapChangerSteps = ({
...
+    const handleResetButton = useCallback(() => {
+        replace(previousValues?.[STEPS] ?? []);
+        setValue(`${tapChanger}.${LOW_TAP_POSITION}`, null);
+        setValue(`${tapChanger}.${STEPS_MODIFIED}`, false);
+        clearErrors(`${tapChanger}.${STEPS}`);
+    }, [clearErrors, previousValues, replace, setValue, tapChanger]);
+
+    const isRegulationRulesModified = useCallback(() => {
+        return modification && areStepsModified;
+    }, [areStepsModified, modification]);
```
Comment (archendel) [inline src/components/dialogs/network-modifications/two-windings-transformer/creation/tap-changer-pane/tap-changer-steps.js:203]: use useMemo instead

File: src/components/dialogs/network-modifications/two-windings-transformer/modification/two-windings-transformer-modification-dialog.js [hunk 2]
```diff
@@ -116,13 +140,70 @@ const TwoWindingsTransformerModificationDialog = ({
...
+            return RATIO_REGULATION_MODES.FIXED_RATIO.id;
+        }
+    };
+
+    const computeHighTapPosition = (steps) => {
+        const values = steps?.map((step) => step[STEPS_TAP]);
+        return Array.isArray(values) && values.length > 0
+            ? Math.max(...values)
+            : null;
+    };
```
Comment (archendel) [inline src/components/dialogs/network-modifications/two-windings-transformer/modification/two-windings-transformer-modification-dialog.js:205]: I've seen this method several times, should we save it in an utils file ?

File: src/components/utils/rhf-inputs/table-inputs/table-numerical-input.js [hunk 3]
```diff
@@ -82,9 +84,10 @@ export const TableNumericalInput = ({
                 style: {
                     fontSize: 'small',
                     color:
-                        previousValue &&
-                        previousValue === value &&
-                        !valueModified
+                        (previousValue !== undefined &&
+                            previousValue === value &&
+                            !valueModified) ||
+                        (handleChange !== undefined && !valueModified)
```
Comment (archendel) [inline src/components/utils/rhf-inputs/table-inputs/table-numerical-input.js:90]: weird usage or handleChange here

**Review by archendel — CHANGES_REQUESTED** [review]

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok  (using this [PR](https://github.com/powsybl/powsybl-network-store/pull/340))

**Review by archendel — APPROVED** [review]

**Pull request merged**